src/Umbraco.Web/UmbracoHelper.cs

```diff
@@ -497,69 +315,37 @@ public IPublishedContent Media(object id)
             return MediaForObject(id);
         }
 
-        private IPublishedContent MediaForObject(object id)
-        {
-            if (ConvertIdObjectToInt(id, out var intId))
-                return _publishedContentQuery.Media(intId);
-            if (ConvertIdObjectToGuid(id, out var guidId))
-                return _publishedContentQuery.Media(guidId);
-            if (ConvertIdObjectToUdi(id, out var udiId))
-                return _publishedContentQuery.Media(udiId);
-            return null;
-        }
+        private IPublishedContent MediaForObject(object id) => _publishedContentQuery.Media(id);
```

Comment:
There is a similar case related to Content (`IPublishedContent ContentForObject(object id)`) where the method was entirely removed as the code block was moved into the [`IPublishedContent Content(object id)`](https://github.com/umbraco/Umbraco-CMS/blob/28bccd4ab2458d4bd9529879479639d43bc9b4b9/src/Umbraco.Infrastructure/PublishedContentQuery.cs#L104) of **PublishedContentQuery.cs**. Regarding Media we also have similar behaviour into https://github.com/umbraco/Umbraco-CMS/blob/28bccd4ab2458d4bd9529879479639d43bc9b4b9/src/Umbraco.Infrastructure/PublishedContentQuery.cs#L149. Could you just clarify what is the right approach here?

Reply:
Should be reintroduced..

The right approach for now is that everything is available on UmbracoHelper, but we should internally not use it. We should use the underlying service. 

The implementations in UmbracoHelper should just wrap the underlying services..

UmbracoHelper cannot be removed for now, as the query builder in templates builds queries  using the UmbracoHelper

Reply:
Got it, thanks 💪

Reply:
Reintroduced it in https://github.com/umbraco/Umbraco-CMS/pull/7744/commits/0fdd0e9d646d019882fb828d4ae9dacb1ebdf57e :slightly_smiling_face: After seeing the tests pass will merge if you don't mind